File: android/app/src/main/java/org/getlantern/lantern/vpn/Provider.java
package org.getlantern.lantern.vpn;

import android.net.VpnService;

// A provider provides the implementation of VPN internals.
interface Provider {
  void run(final VpnService vpnService, final VpnService.Builder builder, final String socksAddr, final String dnsGrabAddr) throws Exception;

  void stop() throws Exception;
}

File: android/app/src/main/kotlin/org/getlantern/lantern/vpn/Provider.kt
package org.getlantern.lantern.vpn

import android.net.VpnService

// A provider provides the implementation of VPN internals.
interface Provider {
    @Throws(Exception::class)
    fun run(
        vpnService: VpnService,
        builder: VpnService.Builder,
        socksAddr: String,
        dnsGrabAddr: String
    )

    @Throws(Exception::class)
    fun stop()
}
